### PR TITLE
Finally fixed the proximity runtime

### PR DESCRIPTION
**Pull request by Cyberboss**
Also fixed some misuse of Move(d)s and destroys

### CODE REVIEW

File: code/__HELPERS/unsorted.dm [hunk 1]
```diff
@@ -1153,7 +1153,7 @@ B --><-- A
 	for(var/B in L)
 		var/turf/C = B
 		LAZYINITLIST(C.proximity_checkers)
-		C.proximity_checkers[A] = TRUE
+		C.proximity_checkers |= A
```
Comment (MrStonedOne) [inline code/__HELPERS/unsorted.dm]: `C.proximity_checkers[A] = TRUE` is faster than `C.proximity_checkers |= A`

**Comment by optimumtact** [comment]
Don't toy with my heart Cyberboss, promise me it's the last time

**Comment by Cyberboss** [comment]
;)

File: code/game/objects/structures/aliens.dm [hunk 2]
```diff
@@ -281,9 +281,9 @@
 							child.Attach(M)
 							break
 
-/obj/structure/alien/egg/Move()
-	remove_from_proximity_list(src, 1)
-	..()
+/obj/structure/alien/egg/Moved(oldloc)
+	remove_from_proximity_list(oldloc, 1)
```
Comment (optimumtact) [inline code/game/objects/structures/aliens.dm]: shouldn't these also add to proximity list in the new loc, or is there something else handling that?

Comment (Cyberboss) [inline code/game/objects/structures/aliens.dm]: Well the only thing that can move it is the singulo right?

Comment (optimumtact) [inline code/game/objects/structures/aliens.dm]: Assuming makes an ass out of u and me

**Pull request merged**